database-migration/src/main/java/uk/ac/ebi/eva/dbmigration/mongodb/ExtractAnnotationFromVariant.java

```diff
@@ -0,0 +1,312 @@
+        Set<String> xrefSet = computeXrefSet(annotationSubdocument, XREFS_FIELD, XREF_ID_FIELD);
+        List<Double> sift = computeMinAndMaxScore(annotationSubdocument, SIFT_FIELD);
+        List<Double> polyphen = computeMinAndMaxScore(annotationSubdocument, POLYPHEN_FIELD);
+
+        Document newAnnotationSubdocument = new Document()
+                .append(VEP_VERSION_FIELD, databaseParameters.getVepVersion())
+                .append(CACHE_VERSION_FIELD, databaseParameters.getVepCacheVersion());
+
+        if (!soSet.isEmpty()) {
+            newAnnotationSubdocument.append(SO_FIELD, soSet);
+        }
+        if (!xrefSet.isEmpty()) {
+            newAnnotationSubdocument.append(XREFS_FIELD, xrefSet);
+        }
+        if (!sift.isEmpty()) {
+            newAnnotationSubdocument.append(SIFT_FIELD, sift);
+        }
+        if (!polyphen.isEmpty()) {
+            newAnnotationSubdocument.append(POLYPHEN_FIELD, polyphen);
+        }
+
+        List<Document> newAnnotationArray = Collections.singletonList(newAnnotationSubdocument);
+
+        Document query = new Document(ID_FIELD, variantDocument.get(ID_FIELD));
+        Bson update = Updates.set(ANNOT_FIELD, newAnnotationArray);
+        return new UpdateOneModel<>(query, update);
+    }
+
+    private static Set<Integer> computeSoSet(Document originalAnnotField, String outerField, String innerField) {
+        Set<Integer> soSet = new TreeSet<>();
+
+        List<Document> cts = (List<Document>) originalAnnotField.get(outerField);
+        if (cts != null) {
+            for (Document ct : cts) {
+                Object sos = ct.get(innerField);
+                if (sos != null) {
+                    soSet.addAll((List<Integer>) sos);
+                }
+            }
+        }
+
+        return soSet;
+    }
+
+    private static Set<String> computeXrefSet(Document originalAnnotField, String outerField, String innerField) {
+        Set<String> xrefSet = new TreeSet<>();
+
+        List<Document> cts = (List<Document>) originalAnnotField.get(outerField);
+        if (cts != null) {
+            for (Document ct : cts) {
+                String xref = ct.getString(innerField);
+                if (xref != null) {
+                    xrefSet.add(xref);
+                }
+            }
+        }
+
+        return xrefSet;
+    }
+
+    private static List<Double> computeMinAndMaxScore(Document originalAnnotField, String field) {
+        Double min = Double.POSITIVE_INFINITY;
+        Double max = Double.NEGATIVE_INFINITY;
+        boolean thereIsAtLeastOneScore = false;
+
+        List<Document> cts = (List<Document>) originalAnnotField.get(CONSEQUENCE_TYPE_FIELD);
+        if (cts != null) {
+            for (Document ct : cts) {
+                Document document = ((Document) ct.get(field));
+                if (document != null) {
+                    Double score = (Double) document.get(SCORE_FIELD);
```

Comment:
the other namings are not perfect, but `SCORE_FIELD = "sc"` is actually the key for the score, so it's not a value. what about field -> scoreType, and keep SCORE_FIELD? otherwise I could rename all the `_FIELD` to `_KEY`